### ISSUE TITLE
Provide better guidance for structured logging migration contributors

### ISSUE THREAD

**Issue by yangjunmyfm192085** [kubernetes/kubernetes#102557]
#### What would you like to be added:
We need to record the files that have been modified in current pr . Prevent multiple PR conflicts
#### Why is this needed:
When reviewing pr about Structured Logging migration, I found that multiple prs conflicted.
For example:
https://github.com/kubernetes/kubernetes/pull/102469 is conflicted with #99534, #99532, #100672, #100699.

**Comment by yuzhiquan** [comment]
May be need an checklist like https://github.com/kubernetes/kubernetes/issues/98976 for diff compoent.

**Comment by serathius** [comment]
Thanks @yangjunmyfm192085 for bringing this up. Coordinating PRs to avoid conflicts would be a one possible solution, but currently we are not planning to start organizing migration of new components until we collect feedback and draw conclusions from Kubelet migration. 

I understand that there is a lot of interest from new contributors to pick migration as an easy task but we don't need to rush as it may lead to making mistakes and needing to repeat some work. 
/cc @ehashman 

I think the better solution for current problem is to provide a better guidance about creating PR and review process. [User guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md) only focuses on implementation side and could provide better information for new contributors that are not members of structured logging working group.

**Comment by serathius** [comment]
I propose to add a section to the top of user guide that specifies:
* How migration is done - We organize migration component by component with time in between to collect feedback. Migration of components that are not marked for it is on best effort and is expected to get delayed review or be rejected (example kubeadm).
* What's the current status of migration - Kubelet was migrated, we are collecting feedback
* How to get review - label with `/wg structured-logging` and cc `@kubernetes/wg-structured-logging-reviews`
* Setup some ground rules that will allow us to reject low quality PRs (some PRs just rename function `Infof` to `InfoS` and nothing else)

/cc @shubheksha @ehashman 
to get some feedback.

@yangjunmyfm192085 would you be interested in updating the guide?

**Comment by serathius** [comment]
Looked through PRs listed in the issue. From those 5 that conflict I would only count one or two with quality good enough that it had chance to be merged, the rest should be closed as it doesn't look like author read the guide at all. That's why we need to first improve our process and quality of PRs, before we scale up.

**Comment by yangjunmyfm192085** [comment]
@serathius yeah, I agree with you, let me  update the user guide firstly.

**Comment by serathius** [comment]
As we are trying to provide better guidance to contributors I think we should reach out to contributex.
/sig contributex

**Comment by serathius** [comment]
Would be great if we could have a separate Pull Request template for structured logging PRs, but I don't know if that's possible.

**Comment by yangjunmyfm192085** [comment]
yeah, It would be great if this works, 
But don't know how to do it

**Comment by serathius** [comment]
/retitle Provide better guidance for structured logging migration contributors

**Comment by serathius** [comment]
@yangjunmyfm192085 I also don't know too much about GitHub PR templating, but we can always ask someone who knows. 
GitHub configuration is kept in `.github` directory which is owned by contributor experience. So let's ask.

@mrbobbytables Is there a way to create a PR template just for structured logging migration?

**Comment by mrbobbytables** [comment]
> @mrbobbytables Is there a way to create a PR template just for structured logging migration?

Sadly no, there can only be a single PR template. =/

Do you have an umbrella or tracking issue (sort of like the checklist mentioned earlier)? A place where folks can "claim" or report on sections they're doing to try and prevent stomping on each other. You can then ask folks to check that issue first before opening a PR.

Other than that, I think the best bet is to update the docs and refer folks to it as often as possible =/

**Comment by serathius** [comment]
@mrbobbytables I provide tracking issue when we are organizing a migration to reach some milestone. Before organizing component migrationwe would need agreement from it's owners. We cannot pick whole k8s as a target as this would take multiple releases with minimal benefit for users vs when we picked Kubelet and migrated all it's logs. 

Still PRs are coming in so I proposed that if contributor really wants some directory to be migrated then they are responsible for avoiding conflicts. Alternative would be to totally forbid independent PRs, but this also doesn't seem good.
